app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -104,6 +107,31 @@ public class LoyaltyCardViewActivity extends CatimaAppCompatActivity implements
     final private TaskHandler mTasks = new TaskHandler();
     Runnable barcodeImageGenerationFinishedCallback;
 
+
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        // Access the SharedPreferences to check if volume keys for navigation are enabled
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+        boolean isVolumeKeysEnabled = sharedPreferences.getBoolean(getString(R.string.settings_key_use_volume_keys), false);
+
+        if (isVolumeKeysEnabled) {
+            if (keyCode == KeyEvent.KEYCODE_VOLUME_UP) {
+                // Navigate to the previous card
+                prevNextCard(false); // Call your existing function for previous card
+                return true; // Indicate that we've handled the event
+            } else if (keyCode == KeyEvent.KEYCODE_VOLUME_DOWN) {
+                // Navigate to the next card
+                prevNextCard(true); // Call your existing function for next card
+                return true; // Indicate that we've handled the event
+            }
+        }
+
+        // If not handled, pass the event to the superclass
+        return super.onKeyDown(keyCode, event);
```

Comment:
Code makes sense, a bit more comments than needed imho but that's personal taste :)

```suggestion
            if (keyCode == KeyEvent.KEYCODE_VOLUME_UP) {
                // Navigate to the previous card
                prevNextCard(false);
                return true;
            } else if (keyCode == KeyEvent.KEYCODE_VOLUME_DOWN) {
                // Navigate to the next card
                prevNextCard(true);
                return true;
            }
        }

        return super.onKeyDown(keyCode, event);
```